File: queues/lunch.py
'''
LeetCode - problem 1700 - Number of Students Unable to Eat Lunch
'''

def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
	a =[0,0]
        for st in students:
		a[st]+=1
        k = 0
        while k < len(sandwiches):
		if a[sandwiches[k]]>0:
                	a[sandwiches[k]]-=1
                else:
                	break
            	k+=1
	return len(sandwiches)-k

File: queues/lunch.cpp
//LeetCode - problem 1700 - Number of Students Unable to Eat Lunch

int countStudents(vector<int>& A, vector<int>& B) {
        int count[] = {0, 0}, n = A.size(), k;
        for (int a : A)
            count[a]++;
        for (k = 0; k < n && count[B[k]] > 0; ++k)
            count[B[k]]--;
        return n - k;
    }

